### PR TITLE
Make: Change make server to make, Change make to make dev

### PR DESCRIPTION
**Pull request by shenli**
@coocood @siddontang

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -132,7 +134,7 @@ interpreter:
 server: parser
 ifeq ($(TARGET), "")
 	rm -rf vendor && ln -s _vendor/vendor vendor
-	@cd tidb-server && $(GO) build -ldflags '$(LDFLAGS)'
+	@cd tidb-server && $(GO) build -ldflags '$(LDFLAGS)' -o ../bin/tidb-server
```
Comment (qiuyesuifeng) [inline Makefile]: How about add `bin` to .gitignore?

Comment (qiuyesuifeng) [inline Makefile]: How about using `$(GO) build -ldflags '$(LDFLAGS)' -o bin/tidb-server tidb-server/main.go`

**Comment by shenli** [comment]
@qiuyesuifeng PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by iamxy** [comment]
LGTM

**Pull request merged**